connector/src/main/java/org/geysermc/connector/entity/living/SquidEntity.java

```diff
@@ -26,10 +26,67 @@
 package org.geysermc.connector.entity.living;
 
 import com.nukkitx.math.vector.Vector3f;
+import com.nukkitx.protocol.bedrock.data.entity.EntityFlag;
+import org.geysermc.connector.entity.Tickable;
 import org.geysermc.connector.entity.type.EntityType;
+import org.geysermc.connector.network.session.GeyserSession;
+import org.geysermc.connector.network.translators.world.block.BlockStateValues;
+
+public class SquidEntity extends WaterEntity implements Tickable {
+
+    private float pitch;
+    private float yaw;
+
+    private float targetPitch;
+    private float targetYaw;
+
+    private boolean inWater;
 
-public class SquidEntity extends WaterEntity {
     public SquidEntity(long entityId, long geyserId, EntityType entityType, Vector3f position, Vector3f motion, Vector3f rotation) {
         super(entityId, geyserId, entityType, position, motion, rotation);
+        this.yaw = rotation.getX();
+    }
+
+    @Override
+    public void tick(GeyserSession session) {
+        checkInWater(session);
```

Comment:
On behalf of Redned:
`The in water check can be optimized
Should be checked only on a move packet being sent
Since I don’t think squids have their position updates on the mc server every single tick`